src/core/gui/sidebar/Sidebar.cpp

```diff
@@ -77,7 +79,11 @@ void Sidebar::buttonClicked(GtkToolButton* toolbutton, SidebarPageButton* button
     }
 }
 
-void Sidebar::addPage(AbstractSidebarPage* page) { this->pages.push_back(page); }
+void Sidebar::addPage(AbstractSidebarPage* page) {
+    const auto ind = std::distance(this->pages.begin(), this->currentPage);
+    this->pages.push_back(page);
+    this->currentPage = this->pages.begin() + ind;
```

Comment:
I would store the page index in that case, it can't be invalidated. Therefore, it is not required to recalculate it.

```suggestion
    this->pages.emplace_back(page);
```